javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/instrumentation/indy/IndyModuleRegistry.java

```diff
@@ -65,6 +69,35 @@ public static InstrumentationModuleClassLoader getInstrumentationClassLoader(
               + " yet");
     }
 
+    if (module instanceof ExperimentalInstrumentationModule) {
+      ExperimentalInstrumentationModule experimentalModule =
+          (ExperimentalInstrumentationModule) module;
+
+      // Opening JPMS modules requires to use a 'witness class' in the target module to get a
+      // reference to the module, which means we have to eagerly load the class.
+      //
+      // However, this code here triggered when the advice is being executed for the first time so
+      // this only creates a very small eager loading that is unlikely to have impact on the
+      // application.
+      //
+      // Also, using a class that is already loaded like the one that is being instrumented or a
+      // related one would increase the likeliness of not having an effect on application class
+      // loading.
+
```

Comment:
this comment is no longer relevant as the witness class got replaced with the module for now

Reply:
Good catch, removed in aa23db016ce9d7fbfec446454c045fbcbded7bd7